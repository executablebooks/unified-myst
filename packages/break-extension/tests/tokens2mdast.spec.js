import { fromMarkdown } from 'mdast-util-from-markdown'

import { mystBreakMmarkExt, mystBreakMdastExt } from '../src/index.js'

const fixtures = [
    '+',
    '+++',
    '+++\n',
    '+++{}',
    '+++{}\n',
    '+++ {"key": "value"}',
]

describe('Convert source to MDAST', () => {
    test.each(fixtures)(`%j`, (content) => {
        const options = {
            extensions: [mystBreakMmarkExt],
            mdastExtensions: [mystBreakMdastExt],
        }
        const mdast = fromMarkdown(content, options)
        expect(mdast).toMatchSnapshot()
    })
})
