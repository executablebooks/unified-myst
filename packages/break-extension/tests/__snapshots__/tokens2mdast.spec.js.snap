// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert source to MDAST "+" 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "checked": null,
          "children": Array [],
          "position": Object {
            "end": Object {
              "column": 2,
              "line": 1,
              "offset": 1,
            },
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "spread": false,
          "type": "listItem",
        },
      ],
      "ordered": false,
      "position": Object {
        "end": Object {
          "column": 2,
          "line": 1,
          "offset": 1,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "spread": false,
      "start": null,
      "type": "list",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 2,
      "line": 1,
      "offset": 1,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Convert source to MDAST "+++ {\\"key\\": \\"value\\"}" 1`] = `
Object {
  "children": Array [
    Object {
      "meta": "{\\"key\\": \\"value\\"}",
      "position": Object {
        "end": Object {
          "column": 21,
          "line": 1,
          "offset": 20,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "mystBreak",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 21,
      "line": 1,
      "offset": 20,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Convert source to MDAST "+++" 1`] = `
Object {
  "children": Array [
    Object {
      "meta": "",
      "position": Object {
        "end": Object {
          "column": 4,
          "line": 1,
          "offset": 3,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "mystBreak",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 1,
      "offset": 3,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Convert source to MDAST "+++\\n" 1`] = `
Object {
  "children": Array [
    Object {
      "meta": "",
      "position": Object {
        "end": Object {
          "column": 4,
          "line": 1,
          "offset": 3,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "mystBreak",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 2,
      "offset": 4,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Convert source to MDAST "+++{}" 1`] = `
Object {
  "children": Array [
    Object {
      "meta": "{}",
      "position": Object {
        "end": Object {
          "column": 6,
          "line": 1,
          "offset": 5,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "mystBreak",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 6,
      "line": 1,
      "offset": 5,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Convert source to MDAST "+++{}\\n" 1`] = `
Object {
  "children": Array [
    Object {
      "meta": "{}",
      "position": Object {
        "end": Object {
          "column": 6,
          "line": 1,
          "offset": 5,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "mystBreak",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 2,
      "offset": 6,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
